### PR TITLE
chore: upgrade nodejs version to v20 and github-actions to latest

### PR DESCRIPTION
**Pull request by jmike**
1. Upgrade Nodejs to v20 in `.nvmrc`
2. Upgrade all github actions to their latest counterparts to resolve the following warnings...

Screenshot 2024-02-25 at 12 28 10 PM

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,32 +17,35 @@ jobs:
...
+    # bypass the action if the PR is a draft
     if: github.event.pull_request.draft == false
     name: build
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+
```
Comment (jmike) [inline .github/workflows/ci.yml:27]: dev: add whitespace for visibility

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,32 +17,35 @@ jobs:
...
           path: '**/node_modules'
           key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json', '.nvmrc') }}
 
-      - uses: actions/cache@v2
+      - uses: actions/cache@v4
+        id: dist
         with:
-          path: '**/build'
-          key: ${{ runner.os }}-build-${{ github.sha }}
+          path: '**/dist'
```
Comment (jmike) [inline .github/workflows/ci.yml:41]: dev: this was wrong all along as the build directory is called `dist` and not `build`

Comment (nikoskalogridis) [reply .github/workflows/ci.yml:41]: Nice catch

**Review by thanoskrg — APPROVED** [review]
Neat and tidy 👌

**Pull request merged**